### PR TITLE
Openai bucket

### PR DESCRIPTION
**Pull request by judegiordano**

### CODE REVIEW

**Comment by flogan** [comment]
lgtm

**Review by flogan — APPROVED** [review]
lgtm!

**Review by cdleveille — APPROVED** [review]

File: src/middleware/index.ts [hunk 1]
```diff
@@ -251,6 +251,20 @@ export const schemas = plugin(async function (app: FastifyInstance) {
 			updated_at: { type: "string" }
 		}
 	});
+	app.addSchema({
+		$id: "openaiChallenge",
```
Comment (cdleveille) [inline src/middleware/index.ts]: Looks like this might be misnamed?

File: src/controllers/image/index.ts [hunk 2]
```diff
@@ -1,42 +1,60 @@
...
+				await users.assertExists({ user_id, server_id });
+				const filename = slugify(`${user_id} ${server_id} ${prompt}.png`);
+				if (filename.length >= 1024) {
+					throw new BadRequestError("input prompt too long");
+				}
 				const response = await OpenAIClient.createImage({
 					prompt,
 					n: 1,
-					size: "512x512"
+					size: "512x512",
```
Comment (cdleveille) [inline src/controllers/image/index.ts]: 1024x1024 is the largest option, I generated one and it is about 3 MB. Do we have enough room on S3 to bump it up?

File: src/controllers/image/index.ts [hunk 3]
```diff
@@ -1,42 +1,66 @@
...
-				const { prompt } = req.body;
-				const response = await OpenAIClient.createImage({
-					prompt,
-					n: 1,
-					size: "512x512"
-				});
-				return { image_url: response.data.data[0].url };
+				const { server_id, user_id } = req.query;
+				await users.assertExists({ user_id, server_id });
+				return await openAiImages.list({ user_id, server_id });
```
Comment (cdleveille) [inline src/controllers/image/index.ts]: Might be good to order by created date desc

**Pull request merged**